### PR TITLE
Introduces writeStyle implementation for basicSupprt for SLD files

### PR DESCRIPTION
**Pull request by KaiVolland**
This introduces a very basic support for wrtiting SLD styles.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "target": "es5",
     "lib": [
       "dom",
-      "es7"
+      "es2017"
```
Comment (chrismayer) [inline tsconfig.json]: Could you please explain, what are the benefits of this change? Thank you!

File: src/SldStyleParser.ts [hunk 2]
```diff
@@ -544,13 +549,393 @@ class SldStyleParser implements StyleParser {
...
+      'Literal': [value]
+    }];
+
+    return sldComparisonFilter;
+  }
+
+  /**
+   * Get the SLD Object (readable with xml2js) from an GeoStyler-Style ComparisonFilter.
+   *
+   * @param {ComparisonFilter} comparisonFilter A GeoStyler-Style ComparisonFilter.
```
Comment (chrismayer) [inline src/SldStyleParser.ts]: => ``@param {Filter} filter A GeoStyler-Style Filter.``

File: src/SldStyleParser.ts [hunk 3]
```diff
@@ -544,13 +549,393 @@ class SldStyleParser implements StyleParser {
...
+          }]
+        }]
+      }
+    };
+  }
+
+  /**
+   * Get the SLD Object (readable with xml2js) from an GeoStyler-Style Rule.
+   *
+   * @param {Rule} rules A GeoStyler-Style Rule.
```
Comment (chrismayer) [inline src/SldStyleParser.ts]: Param description type does not fit the actual function param ``Rule[]``

File: src/SldStyleParser.ts [hunk 4]
```diff
@@ -544,13 +549,393 @@ class SldStyleParser implements StyleParser {
...
+              'Rule': rules
+            }]
+          }]
+        }]
+      }
+    };
+  }
+
+  /**
+   * Get the SLD Object (readable with xml2js) from an GeoStyler-Style Rule.
```
Comment (chrismayer) [inline src/SldStyleParser.ts]: => ``... from GeoStyler-Style Rules``

File: tsconfig.json (hunk 1 above)
Comment (KaiVolland) [reply tsconfig.json]: This allows to use features of es2017. In this case I used the `Object.values` method.

**Comment by KaiVolland** [comment]
Thx for the review. Fixes/Changes incoming...

@marcjansen We currently only support one symbolizer per rule. New Issue --> https://github.com/terrestris/geostyler/issues/85
I'll have a look at the coverage stuff.

File: src/SldStyleParser.ts [hunk 5]
```diff
@@ -544,13 +549,393 @@ class SldStyleParser implements StyleParser {
...
+      mark[0].Fill = [{
+        'CssParameter': [{
+          '_': circleSymbolizer.color,
+          '$': {
+            'name': 'fill'
+          }
+        }]
+      }];
+    }
+    if (circleSymbolizer.strokeColor || circleSymbolizer.strokeColor) {
```
[the remark this answers is not in the document]
Comment (marcjansen) [reply src/SldStyleParser.ts]: Nice catch

File: src/SldStyleParser.ts [hunk 6]
```diff
@@ -544,13 +549,393 @@ class SldStyleParser implements StyleParser {
...
+          'version': '1.0.0',
+          'xsi:schemaLocation': 'http://www.opengis.net/sld StyledLayerDescriptor.xsd',
+          'xmlns': 'http://www.opengis.net/sld',
+          'xmlns:ogc': 'http://www.opengis.net/ogc',
+          'xmlns:xlink': 'http://www.w3.org/1999/xlink',
+          'xmlns:xsi': 'http://www.w3.org/2001/XMLSchema-instance'
+        },
+        'NamedLayer': [{
+          'Name': [
+            'Simple Point'
```
[the remark this answers is not in the document]
Comment (KaiVolland) [reply src/SldStyleParser.ts]: I change this to be more general. We don't habe a name property on the GeoStyler-Style. I think we could add one.

[the remark this answers is not in the document]
Comment (KaiVolland) [reply src/SldStyleParser.ts]: https://github.com/terrestris/geostyler/issues/86

**Comment by KaiVolland** [comment]
I'll opened some new issues for the revealed TODOs:
https://github.com/terrestris/geostyler/issues/87
https://github.com/terrestris/geostyler/issues/88

**Comment by chrismayer** [comment]
Thanks for you ongoing work @KaiVolland! From my point of view this is good to now. Feel free to merge.

**Pull request merged**